### PR TITLE
Dependency Update: use v5.2.1 mocha fork for node 12/LTS support

### LINKED ISSUE #2070: Not usable on Node.js 12
## Issue

Truffle and its sub-packages (`truffle-contract`) are unusable on Node.js 12.

This needs to be **fixed quickly** because it currently makes any package that uses `truffle-contract` **unusable on Node.js 12**.

## Details

Node.js 12 removed some deprecated functions from the V8 add-on API which are used by some of Truffle dependencies.

One of that dependencies is `sha3@1.2.2` which is used by `eth-lib#keccakjs`. To fix this, you need to update all dependencies `yarn.lock`. This should install `sha3@1.2.3` which has fixed compatibility with Node.js 12.

Another dependency with problems is `scrypt.js@0.2.0` which is used by `web3#web3-eth#web3-eth-accounts`. It was fixed in `scrypt.js@0.3.0`. Problem with this is that the fixed version is used only in `web3@1.0.0-beta.55` which is incompatible with Truffle (#1690). To fix this, you need to update Truffle to support `web3@1.0.0-beta.55` which will use the fixed version of `scrypt.js`.

Maybe there are also some other broken dependencies.

## Logs from CI

Error with `sha3`:
https://travis-ci.com/filips123/truffle/jobs/204732468

Error with `scrypt`:
https://travis-ci.com/filips123/EthAvatar.JS/jobs/204728583

## Environment

* Operating system: Ubuntu 14.04 on Travis CI
* Node.js version: 12.3.1
* NPM version: 6.9.0

### PR DESCRIPTION
**Pull request by CruzMolina**
Aims to resolve #2070.

### CODE REVIEW

File: packages/truffle/package.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
   },
   "dependencies": {
     "app-module-path": "^2.2.0",
-    "mocha": "5.2.0",
+    "mocha": "https://github.com/trufflesuite/mocha.git#v5.2.1",
```
Comment (gnidan) [inline packages/truffle/package.json]: This will force users to have git to install Truffle.

Also, don't do this :)

Comment (CruzMolina) [reply packages/truffle/package.json]: Good point. `@truffle/mocha` coming right up. 👏

Comment (gnidan) [reply packages/truffle/package.json]: No but please don't do it this way 🙌

**Comment by CruzMolina** [comment]
Closing since there doesn't seem to be interest in merging this into truffle proper.

File: packages/truffle/package.json (hunk 1 above)
Comment (fainashalts) [reply packages/truffle/package.json:30]: @CruzMolina did can this just say "5.2.1" instead of "5.2.0" or was there a reason you did it this way?

Comment (CruzMolina) [reply packages/truffle/package.json:30]: @fainashalts Currently `5.2.1` only exists as a branch on our fork. Want me to publish it as `@trufflesuite/mocha`?

**Pull request closed without merging**